dut/rocket-chip-dut.wake

```diff
@@ -72,20 +76,31 @@ global def rocketChipDUTMaker dutPlan userArgs =
   def allOutputs = rocketOutputs.getRocketChipGeneratorOutputsAllOutputs
   def topModuleOpt = dutPlan.getRocketChipDUTPlanTestharness.getTestharnessTopModule
   def dts = rocketOutputs.getRocketChipGeneratorOutputsDTS
+
+  # TODO: This should be something that is captured by
+  # RocketChipGeneratorOutputs in rocket-chip.git, since only the Wake rules in
+  # that repo can know what the final output filenames look like.
+  # This can be replaced once we actually add that in rocket-chip.git.
   def dtsJSON =
-    def fileName =
+    def fileNameOpt =
       rocketOutputs
       | getRocketChipGeneratorOutputsInputOptions
-      | getRocketChipGeneratorOptionsConfigNames
-      | catWith "_"
-      | ("_root_.{_}.json")
-    def fileRegex = regExpCat (`([^/]*/)*`, fileName.quote, Nil)
-
-    rocketOutputs
-    | getRocketChipGeneratorOutputsAllOutputs
-    | find (matches fileRegex _.getPathName)
-    | omap getPairFirst
-    | getOrElse "{fileName} file not found".makeError.makeBadPath
+      | getRocketChipGeneratorOptionsBaseFileName
+      | omap ("{_}.json")
+
+    def getDTSJSONForFileName fileName =
+      def fileRegex = regExpCat (`([^/]*/)*`, fileName.quote, Nil)
+
+      rocketOutputs
+      | getRocketChipGeneratorOutputsAllOutputs
+      | find (matches fileRegex _.getPathName)
+      | omap getPairFirst
+      | getOrFail "{fileName} file not found".makeError
+
+    fileNameOpt
+    | getOrFail "dtsJSON not available if outputBaseName not explicitly passed to rocket-chip Generator.scala.".makeError
```

Comment:
nevermind, just read the PR comment

Reply:
Yeah, I would have wanted to make the changes just in rocket-chip, and I'm sort of seeing if I can sneak out that PR now, but there are a lot of things that depend on non-backwards incompatible changes to both api-generator-sifive.git and rocket-chip.git, so I thought that this would be good enough for now.